contracts/strategy/LiquidationStrategy.sol

```diff
@@ -441,6 +501,105 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
+
+        IPermit2(uniswapV3Info[_routerV3].permit2).approve(
+            _tokenIn,
+            address(uniswapV3Info[_routerV3].universalRouter),
+            uint160(_amountIn),
+            uint48(block.timestamp)
+        );
+
+        IUniversalRouter(uniswapV3Info[_routerV3].universalRouter).execute(commands, inputs, block.timestamp);
+
+        ERC20(_tokenIn).safeApprove(uniswapV3Info[_routerV3].permit2, 0);
+
+        uint256 _tokencoinBalanceAfter = ERC20(_tokenOut).balanceOf(address(this));
+
+        receivedAmount = _tokencoinBalanceAfter.sub(_tokencoinBalanceBefore);
+
+        return receivedAmount;
+    }
+
+    // @dev _computeMostProfitablePath should be upgraded/updated once curvePool launches and DEX pool of USDT/FXD will be drained.
+    // an alternative solution can be to involvement of Xswap(UniswapV3)
+    function _computeMostProfitablePath(
+        IUniswapV2Router02 _router,
+        address _collateralToken,
+        uint256 _collateralAmountToLiquidate
+    ) internal view returns (address[] memory, uint256) {
+        // DEX (Collateral -> FXD)
+        address[] memory path1 = new address[](2);
+        path1[0] = _collateralToken;
+        path1[1] = address(fathomStablecoin);
+        uint256 scenarioOneAmountOut = _getDexAmountOut(_collateralAmountToLiquidate, path1, _router);
+
+        // DEX (Collateral -> USDT) -> DEX (USDT -> FXD)
+        address[] memory path2 = new address[](3);
+        path2[0] = _collateralToken;
+        path2[1] = address(usdToken);
+        path2[2] = address(fathomStablecoin);
+        uint256 scenarioTwoAmountOut = _getDexAmountOut(_collateralAmountToLiquidate, path2, _router);
+
+        if (scenarioOneAmountOut >= scenarioTwoAmountOut) {
+            // DEX (Collateral -> FXD)
+            return (path1, scenarioOneAmountOut);
+        } else {
+            // DEX (Collateral -> USDT) -> DEX (USDT -> FXD)
+            return (path2, scenarioTwoAmountOut);
+        }
+    }
+
+    function _getDexAmountOut(
+        uint256 _collateralAmountToLiquidate,
+        address[] memory _path,
+        IUniswapV2Router02 _router
+    ) internal view returns (uint256) {
+        uint256[] memory amounts = _router.getAmountsOut(_collateralAmountToLiquidate, _path);
+        uint256 amountToReceive = amounts[amounts.length - 1];
+        return amountToReceive;
+    }
+
+    function _harvestAndReport() internal view override returns (uint256 _totalAssets) {
+        _totalAssets = asset.balanceOf(address(this));
+    }
+
+    function _encodeSwapExactInputSingle(
+        address recipient,
+        uint256 amountIn,
+        uint256 amountOutMinimum,
+        bytes memory path,
+        bool sourceOfFundsIsMsgSender
+    ) internal pure returns (bytes memory commands, bytes[] memory inputs) {
+        // Command for V3_SWAP_EXACT_IN
+        bytes1 command = 0x00; // Assuming 0x00 is the command for V3_SWAP_EXACT_IN
```

Comment:
You can move this to constants for clarity

Reply:
Ok